lib/docker/util.rb

```diff
@@ -259,21 +286,23 @@ def build_config_header(credentials)
     }
   end
 
-  def glob_all_files(pattern)
-    Dir.glob(pattern, File::FNM_DOTMATCH) - ['..', '.']
+  # do a directory glob that matches .dockerignore behavior
+  # specifically: matched directories are considered a recursive match
+  def dockerignore_compatible_glob(pattern)
+    begin
+      some_dirs, some_files = glob_all_files(pattern).partition { |f| File.directory?(f) }
```

Comment:
I tried to see if we can use `File.fnmatch?` instead of needing to re-glob each time, but apparently the rules for that don't work. `x/**/y` doesn't match `x/y` while it does for `Dir.glob`... That's regardless of the flags passed to `fnmatch?` as well.

Reply:
Does that mean some action is necessary here?  Am I missing a test case?

Reply:
Sorry, nope, was just sharing another avenue that doesn't seem possible right now.